### PR TITLE
ESLint: Add rule to force the usage of `@wordpress/element`

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add a lint rule to force the usage of @wordpress/element.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -488,6 +488,26 @@
...
+        "packages/stories-block/**/*.js",
+        "packages/tinymce-button/**/*.js"
+      ],
+      "rules":{
+        "no-restricted-imports": [
+          "error",
+          {
+            "paths": [
+              {
+                "name": "react",
```
Comment (swissspidy) [inline .eslintrc:507]: Same for react-dom.

Might also want to do the same for encouraging using wordoress/i18n while at it

Comment (spacedmonkey) [reply .eslintrc:507]: I didn't do this, as gutenberg doesn't do this 
https://github.com/WordPress/gutenberg/blob/34fbec666d9b210e94df56246bc208da64d3ad10/.eslintrc.js#L69-L111

Comment (swissspidy) [reply .eslintrc:507]: This is not Gutenberg though. Let‘s add this please

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
FYI there's a merge conflict

File: .eslintrc [hunk 2]
```diff
@@ -516,6 +516,39 @@
...
+      "files": [
+        "packages/**/*.js"
+      ],
+      "excludedFiles": [
+        ".storybook/preview.js",
+        "__mocks__/**/*.js",
+        "packages/activation-notice/**/*.js",
+        "packages/react/**/*.js",
+        "packages/dashboard/src/karma/fixture.js",
+        "packages/story-editor/src/components/canvas/test/useCanvasKeys.js",
```
Comment (swissspidy) [inline .eslintrc]: Why is this added?

That file should just do `import { useRef } from '@web-sories-wp/react';`
